doc/zh-cn/plugins/dynamic-upstream.md

```diff
@@ -0,0 +1,290 @@
+| upstream_host| string | 可选   |        |        | 只在 pass_host 配置为 rewrite 时有效。 |
+| weight       | integer | 可选   |        |        | 根据 weight 值来分配请求的比率。如 weight = 40，说明把40%的请求命中到该上游。 |
+
+## 如何启用
+
+### 灰度发布
+
+根据插件中 upstreams 部分配置的 weight 值，做流量分流。
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "uri": "/index.html",
+    "plugins": {
+        "dynamic-upstream": {
+            "rules": [
+                {
+                    "match": [
+                        {
+                            "vars": [ ]
+                        }            
+                    ],
+                    "upstreams": [
+                        {
+                            "upstream": {
+                                "name": "upstream_A",
+                                "type": "roundrobin",
+                                "nodes": {
+                                    "127.0.0.1:1981":10
+                                },
+                                "timeout": {
+                                    "connect": 15,
+                                    "send": 15,
+                                    "read": 15
+                                }
+                            },
+                            "weight": 40
+                        }
+                    ]
+                }
+            ]
+        }
+    },
+    "upstream": {
+            "type": "roundrobin",
+            "nodes": {
+                "127.0.0.1:1980": 1
+            }
+    }
+}'
+```
+
+### 蓝绿发布
+
+通过请求头获取蓝绿条件(也可以通过请求参数获取)，将插件 upstreams 的 weight 值设置为100，表示所有请求都命中到该upstrean (当 weight 为`0`时，表示所有请求命中 `route` 上默认的 upstream )。
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "uri": "/index.html",
+    "plugins": {
+        "dynamic-upstream": {
+            "rules": [
+                {
+                    "match": [
+                        {
+                            "vars": [
+                                [ "http_new-release","==","blue" ]
+                            ]
+                        }            
+                    ],
+                    "upstreams": [
+                        {
+                            "upstream": {
+                                "name": "upstream_A",
+                                "type": "roundrobin",
+                                "nodes": {
+                                    "127.0.0.1:1981":10
+                                }                                
+                            },
+                            "weight": 100
+                        }
+                    ]
+                }
+            ]
+        }
+    },
+    "upstream": {
+            "type": "roundrobin",
+            "nodes": {
+                "127.0.0.1:1980": 1
+            }
+    }
+}'
+```
+
+### 自定义发布
+
+`match` 中设置多个匹配规则，并设置 `weight` 值。
```

Comment:
we need an example for multiple upstream objects.

Reply:
Only a single upstream is currently supported, and multiple upstreams will be supported in the next "PR".

Reply:
ok